### ISSUE TITLE
on what platforms does "test.rkt" work?

### ISSUE THREAD

**Issue by mbutterick** [takikawa/racket-fontconfig#3]
I’d like to see what it does, but I haven’t gotten it to work on any version of OS X.

**Comment by mbutterick** [comment]
I did get the test working on OS X with the font Charter. Though not Fjalla One or some other test fonts. Not yet clear why that would be so.

**Comment by mbutterick** [comment]
Whoops, never mind — it only worked because I happened to have Charter installed elsewhere as a system font.

**Comment by takikawa** [comment]
@mbutterick yeah, so this test isn't very robust in that it doesn't download the font and then load it. You have to do that manually. I've been thinking about having the test actually download the font and then add it.

**Comment by mbutterick** [comment]
FWIW other than `fc-config-home` I haven’t gotten any `fontconfig` function to work, e.g., —

```racket
(fc-str-list-first (fc-config-get-cache-dirs (fc-config-get-current)))
(fc-str-list-first (fc-config-get-config-dirs (fc-config-get-current)))
(fc-str-list-first (fc-config-get-config-files (fc-config-get-current)))
(fc-str-list-first (fc-config-get-font-dirs (fc-config-get-current)))
```

```
#f
#f
#f
#f
```

**Comment by takikawa** [comment]
The API is a little confusing. You probably actually want `fc-str-list-next` in this case. I think `fc-str-list-first` will set the position in the list to the beginning (for subsequent next calls).

The ffi wrapper could also do a conversion into a Racket list to make it easier too (maybe offering both would make sense?).

**Comment by mbutterick** [comment]
You’re right — that does make a difference:

```racket
(fc-str-list-next (fc-config-get-cache-dirs (fc-config-get-current)))
(fc-str-list-next (fc-config-get-config-dirs (fc-config-get-current)))
(fc-str-list-next (fc-config-get-config-files (fc-config-get-current)))
(fc-str-list-next (fc-config-get-font-dirs (fc-config-get-current)))
```

```
#"/path/to/Racket/font-cache"
#f
#f
#"/path/to/racket/racket/share/fonts"
```

(BTW the `fontconfig` [docs claim](https://www.freedesktop.org/software/fontconfig/fontconfig-devel/fcstrlistfirst.html) that `fc-str-list-first` “Returns the first string” … but OK, I now see its return type is actually `void`, which probably means the description is outdated.)

**Comment by mbutterick** [comment]
It doesn’t change much else, however, because neither `/path/to/Racket/font-cache` nor `/path/to/racket/racket/share/fonts` exist. 

What I’m trying to do is convert font family names to paths on disk. I understand the fandango of doing `fc-name-parse` followed by `fc-config-substitute` and `fc-default-substitute` and `fc-font-match`. I’ve gotten those queries to work, but only when I load the system-level `fontconfig`, which knows about the system fonts, not the Racket-level `fontconfig`, which seems to be empty.

So though `fontconfig` seems to be the right tool for the job, I’m not clear whether these Racket-level bindings can even access the system-level data, or if it is a separate-worlds problem.
